enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,294 @@
+    ```
+
+  - adds [/etc/containers/registries.d/*.yaml](https://github.com/containers/image/blob/main/docs/containers-registries.d.5.md) to allow matching sigstore signatures, for example:
+
+    ```yaml
+    docker:
+      my-registry/image:
+        use-sigstore-attachments: true
+    ```
+
+    Alternatively, we can enable it node wide:
+
+    ```yaml
+    default-docker:
+      use-sigstore-attachments: true
+    ```
+- merges cluster scoped with namespaced ImagePolicy as a json string and update `status` of each ImagePolicy.
+Once an ImagePolicy object gets created/updated, container runtime config controller will call the syncHandler.
+Once an ImagePolicy object gets deleted, container runtime config controller will delete the machine config and run the syncHandler
+
+#### Notes
+
+ImagePolicy is defined as namespaced CRD. `namespace: openshift-config` is used for cluster policy.
+
+### Risks and Mitigations
+
+Risk: The cluster scoped and namespace policies overwrite
+
+Mitigation: MCO container runtime config controller makes sure namespaced policies for an image cannot override global definitions when merging cluster openshift-config policies and namespaced policies into status field of ImagePolicy instances.   
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions [optional]
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+MCO container runtime config controller can add unit tests and e2e tests.
+- unit test: verify the policies from ImagePolicy instances merged correctly and the controller writes correct format policy.json
+- e2e test: verify the MCO writes configuration to the correct location
+
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+#### Removing a deprecated feature
+
+Not applicable.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade expectations:
+In order to use ImagePolicy, an existing cluster will have to make on upgrade.
+
+Downgrade expectations: 
+- Not applicable. 
+  If `N` does not support ImagePolicy CRD, `N+1`supports IamgePolicy. It is not expected that the ImagePolicy related to the failure `N->N+1`. New resources should not be created during `N->N+1`.
+
+
+### Version Skew Strategy
+
+The MCO does not require skew check.
```

Comment:
The following [template point][1] seems worth considering:

> Does this enhancement involve coordinating behavior in the control plane and in the kubelet? How does an n-2 kubelet without this feature available behave when this feature is used?

Are the CRI-O knobs being tuned something that all CRI-Os understand?  And will continue to understand?  Because if the machine-config operator on 4.y sets a setting that 4.(y-2) CRI-O doesn't understand, that seems like it could cause trouble for that node.

[1]: https://github.com/openshift/enhancements/blob/4f18064f4ad9124efec465c59346e37171c0fcc9/guidelines/enhancement_template.md#version-skew-strategy